conda/connection.py

```diff
@@ -107,8 +108,11 @@ def get_binstar_token(url):
             return None
 
     @staticmethod
-    def is_binstar_url(url):
+    def is_binstar_url_needing_token(url):
         urlparts = urlparse(url)
+        token, path = split_token(urlparts.path)
+        if token:  # url already has token
+            return False
         return (urlparts.scheme in ('http', 'https') and
                 any(urlparts.hostname.endswith(bh) for bh in context.binstar_hosts))
```

Comment:
Yeah this still isn't exactly correct.  I spent some time with @dsludwig yesterday trying to get all of this interaction with anaconda-client right, but we need more.  Going to schedule some time for next week.